src/server.cpp

```diff
@@ -38,32 +38,51 @@ RawVector sliceToRaw(grpc_slice slice){
   return out;
 }
 
+void runFunctionIfProvided(List hooks, std::string hook, List params){
```

Comment:
Maybe use CharacterVector or char* for hook instead?  

Since GRPC has it's own string class, I'd like to avoid mixing/matching with std:string also.